### ISSUE TITLE
Mismatch between documented and available attributes

### ISSUE THREAD

**Issue by hacdias** [MinBZK/nl-wallet#17]
Hi!

We're implementing support for the NL Wallet, and I've noticed that there's some mismatch between the documented and the available attributes.

**Documented Attributes**

- Open API Docs: https://github.com/MinBZK/nl-wallet/blob/54dd23c860c7a0fdeb3bca0baa9d9aa454c128a4/documentation/api/wallet-disclosure-components.openapi.yaml
- Relying party docs: https://github.com/MinBZK/nl-wallet/blob/54dd23c860c7a0fdeb3bca0baa9d9aa454c128a4/documentation/relying-party.md#what-a-pid_doctype-looks-like

**Available Attributes**

- Mapping source code: https://github.com/MinBZK/nl-wallet/blob/54dd23c860c7a0fdeb3bca0baa9d9aa454c128a4/wallet_core/wallet/src/document/mapping.rs

**Comparison Table**

| Attribute             | Open API Docs | Relying Party Docs | Mapping Source Code |
| --------------------- | ------------- | ------------------ | ------------------- |
| given_name            | ✅             | ✅                  | ✅                   |
| family_name           | ✅             | ✅                  | ✅                   |
| family_name_prefix    | 🛑             | ✅                  | 🛑                   |
| own_family_name       | ✅             | 🛑                  | 🛑                   |
| given_name_birth      | ✅             | ✅                  | ✅                   |
| family_name_birth     | ✅             | ✅                  | ✅                   |
| gender                | ✅             | ✅                  | ✅                   |
| birth_date            | ✅             | ✅                  | ✅                   |
| age_over_18           | ✅             | ✅                  | ✅                   |
| birth_place           | ✅             | ✅                  | ✅                   |
| birth_city            | ✅             | ✅                  | ✅                   |
| birth_state           | ✅             | ✅                  | ✅                   |
| birth_country         | ✅             | ✅                  | ✅                   |
| has_spouse_or_partner | ✅             | 🛑                  | 🛑                   |
| bsn                   | ✅             | ✅                  | ✅                   |
| nationality           | 🛑             | ✅                  | 🛑                   |
| resident_address      | ✅             | ✅                  | ✅                   |
| resident_street       | ✅             | ✅                  | ✅                   |
| resident_house_number | ✅             | ✅                  | ✅                   |
| resident_postal_code  | ✅             | ✅                  | ✅                   |
| resident_city         | ✅             | ✅                  | ✅                   |
| resident_state        | ✅             | ✅                  | ✅                   |
| resident_country      | ✅             | ✅                  | ✅                   |

**Address Issue**

Besides this, I noticed that if we request a `resident_address`, then the NL Wallet app says that they don't have that information, despite the documentation saying that that is a composition of the other address elements.

**Comment by rubin55** [comment]
Hi @hacdias, thanks for this. This is probably mostly due to changes in our code which have not been reflected in our documentation yet. I've created a ticket in our internal issue tracker and will pick this up. You will probably see the changes with the next public release.

**Comment by rubin55** [comment]
@hacdias, we just pushed v0.2.2 to our GitHub repository which contains small consistency fixes and clarifications with regards to what is implemented, where it comes from, and what isn't. 

You'll find that all elements that exist in `mapping.rs` are now in sync with what is in the OpenAPI spec and in `relying-party.md`.

With regards to `resident_address`, we discovered that the construction of that value is not implented yet; we suggest that you use the parts that constitute it (see `relying-party.md` for details).
